internal/handlers/sqlite/msg_find.go

```diff
@@ -71,6 +71,44 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 		qp.Filter = params.Filter
 	}
 
+	if h.EnableSortPushdown {
+		iter := params.Sort.Iterator()
+		defer iter.Close()
+
+		var key string
+		var order types.SortType
+		var count int
+
+		for ; ; count++ {
+			var k string
+			var v any
+
+			k, v, err = iter.Next()
```

Comment:
I think that's a good example where iterator makes code harder to understand. Checking `params.Sort.Len() == 1` would make this part much simpler